File: 21553 - Sai Nandan D/Week 2/946_Validate Stack Sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = []
        i = 0
        for x in pushed:
            stack.append(x)
            while stack and stack[-1] == popped[i]:
                i += 1
                stack.pop()
                
        return len(stack) == 0

File: 21551-naveen-siva-kumar/week2/946-validate-stack-sequences/946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        
        stack<int> push_st;
        int j=0;
    
        
        for(int i=0;i<pushed.size();i++)
        {
            push_st.push(pushed[i]);
        
            while(push_st.size() > 0 && popped[j] == push_st.top())
            {
                push_st.pop();
            
                j++;
            }
        }
        if(push_st.size()==0)
            return true;
        else 
            return false;

    }
};